app/src/main/java/com/fpvout/digiview/dvr/DVR.java

```diff
@@ -0,0 +1,166 @@
+
+import android.app.Activity;
+import android.media.MediaRecorder;
+import android.os.Environment;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+import android.widget.ImageButton;
+import android.widget.Toast;
+
+import com.fpvout.digiview.R;
+import com.fpvout.digiview.UsbMaskConnection;
+import com.fpvout.digiview.helpers.DataListener;
+import com.fpvout.digiview.helpers.Mp4Muxer;
+import com.fpvout.digiview.helpers.StreamDumper;
+import com.fpvout.digiview.helpers.ThreadPerTaskExecutor;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+public class DVR {
+    private final Activity activity;
+    private boolean recordAmbientAudio;
+    private MediaRecorder recorder;
+    private boolean recording = false;
+    private static DVR instance;
+    private static final String DVR_LOG_TAG = "DVR";
+    private String defaultFolder = "";
+    private String ambietAudio;
+    private String videoFile;
+    private String dvrFile;
+    private String fileName;
+    private StreamDumper streamDumper;
+    private UsbMaskConnection connection;
+    private static Handler updateAfterRecord;
+    public static final String LATEST_THUMB_FILE = "latest.jpeg";
+
+    DVR(Activity activity, boolean recordAmbientAudio, Handler updateAfterRecord, UsbMaskConnection connection){
+        this.activity = activity;
+        this.connection = connection;
+        this.recordAmbientAudio = recordAmbientAudio;
+        this.updateAfterRecord = updateAfterRecord;
+        defaultFolder =  Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES) + "/" + this.activity.getApplicationInfo().loadLabel(this.activity.getPackageManager()).toString();
+        streamDumper = new StreamDumper(activity, defaultFolder);
+    }
+
+    public static DVR getInstance(Activity context, boolean recordAmbientAudio, Handler updateAfterRecord, UsbMaskConnection connection){
+        if (instance == null) {
+            instance = new DVR(context, recordAmbientAudio, updateAfterRecord, connection);
+        }
+        return instance;
+    }
+
+    public void init() throws IOException {
+        repairNotFinishedDVR();
+        recorder = new MediaRecorder();
+        recorder.setAudioSource(MediaRecorder.AudioSource.MIC);
+        recorder.setOutputFormat(MediaRecorder.OutputFormat.AAC_ADTS);
+        recorder.setAudioEncoder(MediaRecorder.AudioEncoder.AAC);
+    }
+
+    private void  repairNotFinishedDVR(){
+        File dvrFolder = new File(defaultFolder);
+        if (dvrFolder.exists()) {
+            File[] files = dvrFolder.listFiles();
+            for (int i = 0; i < files.length; ++i) {
+                File file = files[i];
+                if (file.getAbsolutePath().endsWith(".h264")) {
+                    File ambientAudio = new File(file.getAbsolutePath().replace(".h264", ".aac"));
+                    if (ambientAudio.exists()) {
```

Comment:
Consider muxing even if the ambient audio is not existing, to at least have a useable video file.

Reply:
great idea !

Reply:
Yes we can do that. No Problem 10 Mins of awesome sound ... you never know whats beeing recorded :D